builtin/mainmenu/tab_online.lua

```diff
@@ -346,12 +346,18 @@ local function main_button_handler(tabview, fields, name, tabdata)
 		return true
 	end
 
-	if (fields.btn_mp_login or fields.key_enter)
-			and fields.te_address ~= "" and fields.te_port then
+	local te_port_number = tonumber(fields.te_port)
+	-- te_port_number will be nil if input is not a number
+	-- '%1' gets fractional part, which is zero for integers
+	-- '1/0' is infinity, which a port cannot be
+	local host_filled = (fields.te_address ~= "") and te_port_number
+		and (te_port_number > 0) and (te_port_number ~= 1/0) and (te_port_number % 1 == 0)
```

Comment:
At this point it might be quicker to use `string.match` for higher reliability:
```Lua
local t = {
	"  32095 ",
	"32095",
	"3nan",
	"inf",
	"-325"
}

for i, v in pairs(t) do
	local num = string.match(v or "", "^%s*(%d+)%s*$")
	print(v, num)
end
```

Reply:
I prefer the explicit number checking, personally more readable to me.
But if two of you want it then sure.
This won't check for `!=0` though, so still have to do that.

Reply:
actually I suppose `^%s*[1-9][0-9]*%s*$` would do that.
we end up with
```lua
local host_filled = (fields.te_address ~= "") and fields.te_port:match("^%s*[1-9][0-9]*%s*$")
local te_port_number = tonumber(fields.te_port)
```
unless you want me to un-do extracting the `tonumber` and put it pack into being repeated?

Reply:
oh, just realised you said reliability not readability, that makes more sense